docs/plugins/gatsby-remark-relative-linker/index.js

```diff
@@ -1,28 +1,19 @@
 const visit = require("unist-util-visit");
-module.exports = ({ markdownAST }) => {
+const updateLinkText = require("./functions");
+
+module.exports = ({ markdownAST, markdownNode }) => {
   visit(markdownAST, "link", (node) => {
     if (
       !node.url.startsWith("/") &&
       !node.url.startsWith("#") &&
       !node.url.startsWith("mailto:") &&
       !/^https?:\/\//.test(node.url)
     ) {
-      let foundNoPrefix = node.url.match(/^[\w-_]+/);
-      let foundDepth1 = node.url.match(/^\.\//);
-      let foundDepth2 = node.url.match(/^\.\.\//);
-      let foundDepth3 = node.url.match(/^\.\.\/\.\.\//);
-
-      if (foundDepth1) {
-        node.url = node.url.replace(/^\.\//, "../");
-      } else if (foundDepth2) {
-        node.url = node.url.replace(/^\.\.\//, "../../");
-      } else if (foundDepth3) {
-        node.url = node.url.replace(/^\.\.\/\.\.\//, "../../../");
-      }
+      const isIndex =
+        markdownNode.fileAbsolutePath.includes("index.md") ||
+        markdownNode.fileAbsolutePath.includes("README.md");
 
-      if (foundNoPrefix) {
-        node.url = "../" + node.url;
-      }
+      node.url = updateLinkText(node.url, isIndex);
```

Comment:
would this work if the function signature does not match? if it does not work, this indicates the necessity of including docs/plugins/gatsby-remark-relative-linker/test.js to be run in CI so that we can capture this during code review